ts/components/NavBarLabel.tsx

```diff
@@ -50,20 +52,55 @@ const styles = StyleSheet.create({
   }
 });
 
+const computeAccessibilityLabel = (
+  section: string,
+  order: number,
+  unread?: number
+) =>
+  typeof unread === "undefined"
+    ? I18n.t("navigation.accessibility", {
+        section,
+        order
+      })
+    : I18n.t("navigation.accessibilityWithBadge", {
+        section,
+        order,
+        unread
+      });
+
 /**
  * This Component is used to Render the Labels of the bottom navbar of the app
  * translated in the preferred locale if it was selected
  * @param props
  */
 const NavBarLabel: React.FunctionComponent<Props> = (props: Props) => {
-  const { options, routeName, preferredLanguage } = props;
+  const {
+    options,
+    routeName,
+    preferredLanguage,
+    messagesUnread,
+    transactionsNumUnread
+  } = props;
   const locale: Locales = preferredLanguage.fold(I18n.locale, l => l);
   const label = getLabel(routeName, locale);
   const maybeOrder = fromNullable(routeOrder.get(routeName as Routes));
   const isSelected = options.focused
     ? `${I18n.t("navigation.selected")}, `
     : "";
 
+  const unreadMessagesMap = {
+    [ROUTES.MESSAGES_NAVIGATOR]: messagesUnread.length,
+    [ROUTES.WALLET_HOME]: transactionsNumUnread
+  };
+
+  const computeUnreadMessages = unreadMessagesMap[routeName] || undefined;
```

Comment:
🤓 **suggestion**: `computeUnreadMessages` sounds more like a function because of the imperative tense - I'd use participle or adjectives for variables like `computedUnreadMessages` instead